### ISSUE TITLE
Error marker when null analysis enabled

### ISSUE THREAD

**Issue by snjeza** [eclipse-jdt/eclipse.jdt.core#854]
A related issue - https://github.com/redhat-developer/vscode-java/issues/3002

Steps to reproduce:
- import[ test project](https://github.com/snjeza/vscode-test/raw/master/testissue3002.zip)
- you will get this error
```
Type mismatch: cannot convert from Class<App.MyInit> to Class<? extends Init<? extends Configuration>>
```

**Comment by stephan-herrmann** [comment]
This problem is covered by #1067

**Comment by stephan-herrmann** [comment]
New test has been disabled via #1079 to be re-fixed via #1077

**Comment by stephan-herrmann** [comment]
Resolved via #1077
